android-components/components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/ThumbnailsMiddleware.kt

```diff
@@ -56,6 +51,19 @@ class ThumbnailsMiddleware(
             }
         }
 
-        next(action)
+        when (action) {
+            is ContentAction.UpdateThumbnailAction -> {
+                // Store the captured tab screenshot from the EngineView when the session's
+                // thumbnail is updated.
+                thumbnailStorage.saveThumbnail(action.sessionId, action.thumbnail)
+            }
+            is ContentAction.RemoveThumbnailAction -> {
+                thumbnailStorage.deleteThumbnail(action.sessionId)
+            }
+            else -> {
+                // Do not let the thumbnail actions continue through to the reducer.
+                next(action)
+            }
+        }
```

Comment:
nit: Could we have used `return` for these two actions and merged two `when`s? Not sure which looks more readable though 😄 (seems almost similar to me)

Reply:
Oh yes, that is a lot nicer to read actually! The tests pass too - thanks!